addons/point_of_sale/views/pos_payment_views.xml

```diff
@@ -11,8 +11,14 @@
                         <field name="pos_order_id"/>
                         <field name="amount" readonly="0"/>
                         <field name="payment_method_id"/>
-                        <field name="card_type" invisible="not card_type"/>
-                        <field name="cardholder_name" invisible="not cardholder_name"/>
+                        <field name="payment_method_payment_mode" invisible="not payment_method_payment_mode"/>
+                        <field name="card_type" invisible="not card_type or card_no == 'null'"/>
+                        <field name="card_brand" string="Card's Brand" invisible="not card_brand or card_no == 'null'"/>
+                        <field name="card_no" invisible="not card_no or card_no == 'null'"/>
+                        <field name="cardholder_name" invisible="not cardholder_name or card_no == 'null'"/>
+                        <field name="payment_method_issuer_bank" invisible="not payment_method_issuer_bank"/>
+                        <field name="payment_method_authcode" invisible="not payment_method_authcode"/>
+                        <field name="payment_ref_no" invisible="not payment_ref_no"/>
                         <field name="transaction_id" invisible="not transaction_id"/>
```

Comment:
You can put this line above the new lines to conserve the old format

Reply:
I made this sequence after confirming with PO mog here, he wants details should show in a sequence like:
order details, amount, payment method, card details if available then all other details.